shared/javascripts/event_emitter.js

```diff
@@ -25,40 +25,46 @@ if (Privly === undefined) {
 
   /**
    * Add an event listener.
-   * 
+   *
    * @param  {String}   ev Event name
    * @param  {Function} listener
    */
   Privly.EventEmitter.prototype.on = function (ev, listener) {
+    if(ev === undefined) {
+      return false;
+    }
     if (this.__listeners === undefined) {
       this.__listeners = {};
     }
     if (!this.__listeners.hasOwnProperty(ev)) {
       this.__listeners[ev] = [];
     }
-    if (this.__listeners[ev].indexOf(listener) === -1) {
+    if (listener !== undefined && this.__listeners[ev].indexOf(listener) === -1) {
       this.__listeners[ev].push(listener);
     }
   };
 
   /**
    * Remove an event listener.
-   * 
+   *
    * @param  {String} ev Event name
    * @param  {Function} listener
    */
   Privly.EventEmitter.prototype.off = function (ev, listener) {
+    if(ev === undefined) {
```

Comment:
You are changing the signature of the function to make it easier to test. This is generally fine, but at this point the two potential return values are `false` and `undefined`, which can introduce problems in the boolean interpretation of the return value. It might be better to throw an error in the event of invalid input, and test for the error from the unit test.